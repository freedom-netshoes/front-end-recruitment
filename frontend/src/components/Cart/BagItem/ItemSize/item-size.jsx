import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import CSSModules from 'react-css-modules'

import styles from './item-size.less'

@CSSModules(styles)
export default class ItemSize extends Component {
  static defaultProps = {
    sizes: ''
  };

  static propTypes = {
    sizes: PropTypes.string
  };

  render () {
    return (
      <p styleName='size'>{this.props.sizes}</p>
    )
  }
}
