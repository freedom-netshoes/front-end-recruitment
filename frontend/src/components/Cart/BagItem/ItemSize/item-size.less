@import '../../../../styles/_less-variables.less';
@import '../../../../styles/_less-mixins.less';
.size {
    .fontLight();
    color: @lighterGray;
    font-size: 12px;
    margin: 7px 0;
}