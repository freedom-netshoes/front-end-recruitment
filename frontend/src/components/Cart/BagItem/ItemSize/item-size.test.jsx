/* global describe, it, expect */
import React from 'react'
import ItemSize from './item-size'
import { shallow } from 'enzyme'

describe('Testing <ItemSize /> component', () => {
  it('renders the ItemSize component', () => {
    const component = shallow(<ItemSize />)
    expect(component).toHaveLength(1)
  })

  it('Testing snapshop in <ItemSize >', () => {
    let component = shallow(<ItemSize sizes={['S', 'G', 'GG', 'GGG'].join(' | ')} />)
    expect(component).toMatchSnapshot()
  })
})
