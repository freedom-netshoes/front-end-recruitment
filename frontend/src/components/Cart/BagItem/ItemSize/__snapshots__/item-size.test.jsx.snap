// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ItemSize /> component Testing snapshop in <ItemSize > 1`] = `
<p
  className="size"
>
  S | G | GG | GGG
</p>
`;
